### PR TITLE
mon: use CephOsdName for osd args instead of CephInt

### PR DESCRIPTION
**Pull request by tchaikov**
this pr fixes the "array of non-strings" issue in #6085.

### CODE REVIEW

**Comment by tchaikov** [comment]
test failed due to test/encoding/check-generated.sh
```
2 bluestore_wal_transaction_t
/tmp/typ-DsevwL6jH /tmp/typ-xpbNCcf7v differ: byte 60, line 5
**** bluestore_wal_transaction_t test 2 dump_json check failed ****
   ./ceph-dencoder type bluestore_wal_transaction_t select_test 2 dump_json > /tmp/typ-DsevwL6jH
   ./ceph-dencoder type bluestore_wal_transaction_t select_test 2 encode decode dump_json > /tmp/typ-xpbNCcf7v
   diff /tmp/typ-DsevwL6jH /tmp/typ-xpbNCcf7v
/tmp/typ-DsevwL6jH /tmp/typ-oTNBjA1Zd differ: byte 60, line 5
**** bluestore_wal_transaction_t test 2 copy dump_json check failed ****
   ./ceph-dencoder type bluestore_wal_transaction_t select_test 2 dump_json > /tmp/typ-DsevwL6jH
   ./ceph-dencoder type bluestore_wal_transaction_t select_test 2 copy dump_json > /tmp/typ-xpbNCcf7v
   diff /tmp/typ-DsevwL6jH /tmp/typ-xpbNCcf7v
/tmp/typ-DsevwL6jH /tmp/typ-eehbuqWan differ: byte 60, line 5
**** bluestore_wal_transaction_t test 2 copy_ctor dump_json check failed ****
   ./ceph-dencoder type bluestore_wal_transaction_t select_test 2 dump_json > /tmp/typ-DsevwL6jH
   ./ceph-dencoder type bluestore_wal_transaction_t select_test 2 copy_ctor dump_json > /tmp/typ-xpbNCcf7v
   diff /tmp/typ-DsevwL6jH /tmp/typ-xpbNCcf7v
/tmp/typ-DsevwL6jH /tmp/typ-xpbNCcf7v differ: byte 25, line 1
**** bluestore_wal_transaction_t test 2 binary reencode check failed ****
   ./ceph-dencoder type bluestore_wal_transaction_t select_test 2 encode export /tmp/typ-DsevwL6jH
   ./ceph-dencoder type bluestore_wal_transaction_t select_test 2 encode decode encode export /tmp/typ-xpbNCcf7v
   cmp /tmp/typ-DsevwL6jH /tmp/typ-xpbNCcf7v
```

**Comment by tchaikov** [comment]
the test failure is fixed in #7168.

File: src/common/cmdparse.cc [hunk 1]
```diff
@@ -155,16 +155,35 @@ cmdmap_from_json(vector<string> cmd, map<string, cmd_vartype> *mapp, stringstrea
...
 	  }
-	  (*mapp)[it->first] = outv;
+	  // if an empty array is acceptable, the caller should always check for
+	  // vector<string> if the expected value of "vector<int64_t>" in the
+	  // cmdmap is missing.
+	  if (out_ints.empty())
+	    (*mapp)[it->first] = out_strs;
+	  else
+	    (*mapp)[it->first] = out_ints;
 	}
```
Comment (aclamk) [inline src/common/cmdparse.cc]: Would it not look better if you split look int 2 separate loops: for int64 and for strings?

Comment (tchaikov) [inline src/common/cmdparse.cc]: yeah, done.

**Comment by tchaikov** [comment]
changes

- rebased against master
- addressed @aclamk 's comment.

**Comment by tchaikov** [comment]
@liewegas is it still relevant? or this approach is overkill for using CephOsdName?

**Comment by liewegas** [comment]
I think it's still relevant!  I think the main thing missing from this PR now is a bunch of additions to qa/workunit/cephtest/test.sh to ensure that both ints and osdnames work for these commands.

**Comment by tchaikov** [comment]
thanks sage, will come back with more coverage.

**Pull request merged**